handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -573,12 +576,11 @@ public SSLEngine engine() {
      * @return the protocol name or {@code null} if application-level protocol has not been negotiated
      */
     public String applicationProtocol() {
-        SSLSession sess = engine().getSession();
-        if (!(sess instanceof ApplicationProtocolAccessor)) {
-            return null;
+        SSLSession sess = engine.getSession();
+        if (sess instanceof ApplicationProtocolAccessor) {
+            return ((ApplicationProtocolAccessor) sess).getApplicationProtocol();
         }
-
```

Comment:
should we try to use http://download.java.net/java/jdk9/docs/api/javax/net/ssl/SSLEngine.html#getApplicationProtocol-- here?